### DISCUSSION TITLE
Detect Builder without a factory method

### DISCUSSION THREAD

**Discussion by rmschots** [mapstruct/mapstruct#3724] [Ideas]
Currently the `DefaultBuilderProvider` only detects builders if there is a static builder factory method available.
As I'm trying to use Wire (https://github.com/square/wire), and it only generates the inner builder class, I'd like to have mapstruct detect and use that builder class out of the box.

I brought us this issue at the Wire repository (https://github.com/square/wire/issues/3073), but the introduction of a static factory method for the builder got rejected.

I think it's either way more appropriate to add support in mapstruct out of the box.
I created an SPI that adds this support, but I prefer to have this baked into mapstruct.
I could open a PR if needed.

an example of target that cannot be used for mapping:
```
public class SomeTarget {
    private final String someField;

    private SomeTarget(String someField) {
        this.someField = someField;
    }

    public static class SomeTargetBuilder {
        private String someField;

        public SomeTargetBuilder someField(String someField) {
            this.someField = someField;
            return this;
        }

        public SomeTarget build() {
            return new SomeTarget(this.someField);
        }
    }
}
```
Trying to generate a mapper would currently produce the following error: `SomeTarget does not have an accessible constructor.`

**Comment by filiphr** [comment]
Hey @rmschots, I don't think that it'll be too difficult to support Wire with a custom builder provider and looking at the code that wire generates it should be relatively trivial to support this.

Feel free to propose your PR and we can see how it looks like. Please also add integration tests for it as well
